_posts/2016-05-12-publish-subscribe-model-in-kafka.md

```diff
@@ -0,0 +1,123 @@
+---
+layout: post
+name: Publish-subscribe-model-in-Kafka
+title: 'Publish-subscribe model in Kafka'
+date: 2016-05-12 00:20:00 +00:00
+author: Felipe Fernández
+image:
+    src: /assets/img/custom/blog/kafka.jpg
+    attribution:
+      text: Kafka Museum by Priit Tammets
+      href: https://www.flickr.com/photos/tammets/2116102744
+tags:
+- publish-subscribe
+- kafka
+- messaging
+
+---
+
+This is the fourth post in the series about integrating sync clients with async systems ([1, ](http://codurance.com/2016/04/28/async-systems-with-sync-clients/)[2, ](http://codurance.com/2016/04/30/akka-basics/)[3](http://codurance.com/2016/05/10/finite-state-machines-with-akka/)). Here we'll try to understand how [Kafka](http://kafka.apache.org/) works in order to leverage properly its publish-subscribe implementation.
+
+
+## Kafka concepts
+
+According to [official documentation](http://kafka.apache.org/documentation.html#introduction):
+
+> Kafka is a distributed, partitioned, replicated commit log service. It provides the functionality of a messaging system, but with a unique design.
+
+Kafka runs as a cluster and the nodes are called brokers. Brokers can be leaders or replicas to provide high-availability and fault tolerance. Brokers are in charge of partitions, being the distribution unit where messages are stored. Those messages are ordered and they're accessible by offset. A set of partitions forms a topic, being a feed of messages. A partition can have different consumers, and they access to the messages using its own offset. Producers publish messages into Kafka topics.
```

Comment:
Would help if you showed a diagram with a basic topology of a Kafka setup (cluster, broker, communication with other entities) - which would explain some of the things you mentioned here and below.